### PR TITLE
fix(gateway): Register schema changes for unmanaged configs

### PR DESCRIPTION
**Pull request by cmschuetz**
When using an unmanaged gateway configuration with `experimental_pollInterval` set, `onSchemaChange` callbacks don't get registered because the function short-circuits if it detects an unmanaged config.  This fix ensures that apollo server is notified of gateway schema updates.

### CODE REVIEW

File: packages/apollo-gateway/src/__tests__/gateway/lifecycle-hooks.test.ts [hunk 1]
```diff
@@ -177,4 +179,31 @@ describe('lifecycle hooks', () => {
     );
     consoleSpy.mockRestore();
   });
+
+  it('registers schema change callbacks when experimental_pollInterval is set for unamanged configs', async () => {
```
Comment (trevor-scheer) [inline packages/apollo-gateway/src/__tests__/gateway/lifecycle-hooks.test.ts]:
```suggestion
  it('registers schema change callbacks when experimental_pollInterval is set for unmanaged configs', async () => {
```

**Comment by cmschuetz** [comment]
@trevor-scheer thanks for the quick review, added a changelog entry

**Comment by trevor-scheer** [comment]
Thanks so much @cmschuetz!

**Comment by abernix** [comment]
This fixes the issue demonstrated in the reproduction provided by @mark-beeby
 in https://github.com/apollographql/apollo-server/pull/3562#issuecomment-559493848. 🎉 

It's been released in 2.9.15.

Thanks so much for working on this fix @cmschuetz.  We really appreciate your contribution! 🙌

**Pull request merged**